### ISSUE TITLE
Downloads get stuck at like 80% and time out

### ISSUE THREAD

**Issue by unixero** [ppareit/swiftp#252]
I'm using version 3.1 from f-droid on Android 14.0.0 on a Google Pixel 7a.
On my Linux PC I download pictures from my phone using FileZilla 3.52.2, but the same happens using netkit-ftp 0.17-34.1.1 from command line, both in passive mode.
AFWall+ is enabled, FTP server is allowed only local network access over wifi.

Whenever I download a file it quickly gets to like 80% and gets stuck there until FileZilla times out and reconnects, then it continues donwloading and gets to 85%... then again at 90%... but never really finishes downloading.

It doesn't always get stuck at the same point, but it's usually between 75% and 90%. Every time at the retry (continuing from the point where it stopped) it makes a much smaller transfer than the first time, so it never reaches 100%. FileZilla aborts automatically after 3 retries.

If I download a much bigger file, it still gets to like 80% (which is a lot more in absolute number of bytes transferred).

I have no issues downloading form another server (e.g. proftpd running on a raspberry pi).

**Comment by unixero** [comment]
Just to add more (potentially useful) info: the issue is not reproducible with my old phone (Asus Zenfone 3 max 5.5) running Android 8.1. Same version of ftp server, same PC with same version of FileZilla, same permissions on AFWall+. Probably caused newer Android versions?

**Comment by Xavron** [comment]
The f-droid version is *really old* right now and not really a good idea to base an issue on. The newer source code here on git is already fully compatible with Android 14 (and much more) over that f-droid version.

Although, a timeout in the middle of a transfer should be a network stability issue (if bytes were transferred to the client and the client created a file and that file is not a 0 byte file). Provided that the older sdk target in that version is doing something that is allowed to work on internal storage.

_*Network stability issues can result on one device, difficult to notice on multiple devices, from a device, be a device issue, etc. As an example, my one router AP will randomly kick my Android 8 device off wifi but not do that to other devices. Another AP doesn't do this (this is a router bug). This can cause an attempt to resume the download. That can result in other things happening with the ftp server and then fail due to incompatibility of the older app version with Android 14 or a code issue such as one I've deal with for scoped. It can even just be plain wrong if the router badly wrecked things. Its complicated._

**Comment by unixero** [comment]
Thanks for posting that comment about the router. I've realized the issue is not reproducible if I set a fixed IP address. It's confusing because only the ftp server is affected, not other apps, but it is indeed solved this way. Closing this issue as it is invalid.

BTW if the f-droid version is really old and you have much better features in the newer source... then probably it's time to make a new tag.  Even if you wanted users to compile from source, 3.1.0 is still the latest tagged version.
